services/platform/platform.go

```diff
@@ -0,0 +1,199 @@
+			// continue with next loop iteration
+		case <-ctx.Done():
+			return
+		}
+
+		nCtx, cancel := context.WithTimeout(ctx, timeout)
+		err := s.RefreshSession(nCtx)
+		if err != nil && err != errNoActiveSessions {
+			s.l.Warnf("Failed to refresh session, reason: %+v.", err)
+		}
+		cancel()
+	}
+}
+
+// SignUp creates new Percona Platform user with given email and password.
+func (s *Service) SignUp(ctx context.Context, email, password string) error {
+	cc, err := s.getConnection(ctx)
+	if err != nil {
+		return errors.Wrap(err, "failed establish connection with Percona")
+	}
+	defer cc.Close() //nolint:errcheck
+
+	_, err = api.NewAuthAPIClient(cc).SignUp(ctx, &api.SignUpRequest{Email: email, Password: password})
+	if err != nil {
+		return errors.Wrap(err, "failed to sign up user")
+	}
+
+	return nil
+}
+
+// SignIn checks Percona Platform user authentication and creates session.
+func (s *Service) SignIn(ctx context.Context, email, password string) error {
+	cc, err := s.getConnection(ctx)
+	if err != nil {
+		return errors.Wrap(err, "failed establish connection with Percona")
+	}
+	defer cc.Close() //nolint:errcheck
+
+	resp, err := api.NewAuthAPIClient(cc).SignIn(ctx, &api.SignInRequest{Email: email, Password: password})
+	if err != nil {
+		return errors.Wrap(err, "failed to sign in user")
+	}
+
+	err = s.db.InTransaction(func(tx *reform.TX) error {
+		params := models.ChangeSettingsParams{SessionID: resp.SessionId, Email: email}
+		_, err := models.UpdateSettings(tx.Querier, &params)
+		return err
+	})
+	if err != nil {
+		return errors.Wrap(err, "failed to save session id")
+	}
+
+	return nil
+}
+
+// RefreshSession resets session timeout.
+func (s *Service) RefreshSession(ctx context.Context) error {
+	settings, err := models.GetSettings(s.db)
+	if err != nil {
+		return err
+	}
+
+	if settings.SaaS.SessionID == "" {
+		return errNoActiveSessions
+	}
+
+	cc, err := s.getConnection(ctx)
+	if err != nil {
+		return errors.Wrap(err, "failed establish connection with Percona")
+	}
+	defer cc.Close() //nolint:errcheck
+
+	md := metadata.New(map[string]string{"authorization": authType + " " + settings.SaaS.SessionID})
+	_, err = api.NewAuthAPIClient(cc).RefreshSession(metadata.NewOutgoingContext(ctx, md), &api.RefreshSessionRequest{})
+	if err != nil {
+		return errors.Wrap(err, "failed to refresh session")
+	}
+
+	return nil
+}
+
+func (s *Service) getConnection(ctx context.Context) (*grpc.ClientConn, error) {
+	host, _, err := net.SplitHostPort(s.host)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to set checks host")
+	}
+	tlsConfig := tlsconfig.Get()
+	tlsConfig.ServerName = host
+
+	opts := []grpc.DialOption{
+		// replacement is marked as experimental
+		grpc.WithBackoffMaxDelay(timeout), //nolint:staticcheck
+
+		grpc.WithBlock(),
+		grpc.WithUserAgent("pmm-managed/" + version.Version),
+		grpc.WithTransportCredentials(credentials.NewTLS(tlsConfig)),
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, timeout)
```

Comment:
@artemgavrilov We can remove that one then

Reply:
Actually, no.
We should have two different timeout: dial timeout (5 seconds) and total timeout (like 10 seconds). We should pass the second timeout to all Auth Service calls, not only to refresh, but also to SignIn/SignUp

Reply:
Hm, ok. But shouldn't that timeout be controlled by managed API client? I mean if UI creates request with timeout 30s why we limit it 10s?

Reply:
UI doesn't actually control timeout – we don't document `Grpc-Timeout` header in our exposed JSON API (and I don't think we ever tested it). Timeout in Go code will allow UI to get "good" error code instead of generic 504 from nginx.

Reply:
And we _almost_ can set timeout in interceptor – except that we have several special cases where we need higher timeout: logs.zip, pmm-agent protocol, some QAN APIs, etc. So for now we stuck with per-method or per-service timeouts.

Reply:
Ok, thanks for explanation. Will add timeouts 👌

Reply:
@AlekSi take a look at the last commit, please. I think timeouts should be defined on the caller side, not inside methods, so I defined them in `server.go`